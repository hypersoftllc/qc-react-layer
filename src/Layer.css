.Layer {
  /* Backup */
  isolation: isolate;
  /* Backup */
  -webkit-overflow-scrolling: touch;
  /* Establish a stacking context in any browser. */
  position: relative;
  z-index: 0;
}
