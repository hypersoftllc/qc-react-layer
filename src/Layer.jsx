import PT from 'prop-types'
import React from 'react'
import Block from '@qc/react-block'


function Layer(props) {
  const domProps = {
    ...props,
    className: `${props.compClassName} ${props.className}`.trim(),
  }
  delete domProps.children
  delete domProps.compClassName
  return (
    <Block {...domProps}>
      {props.children}
    </Block>
  )
}

Layer.defaultProps = {
  className: '',
  compClassName: 'Layer',
}

Layer.propTypes = {
  children: PT.node,
  className: PT.string,
  compClassName: PT.string,
}

export default Layer
