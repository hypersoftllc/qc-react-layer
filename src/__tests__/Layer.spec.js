/* eslint-env jest */
/* eslint padded-blocks: "off" */

import React from 'react'
import TestRenderer from 'react-test-renderer'

import Layer from '../Layer'


describe('Layer', () => {

  it('should be a function', () => {
    expect(typeof Layer).toBe('function')
  })

  it('should be a functional component', () => {
    let comp

    comp = TestRenderer.create(
      <Layer/>
    )

    // Functional components don't have instances.
    expect(comp.getInstance()).toBe(null)
  })

  describe('rendered', () => {

    describe('with no props', () => {

      it('should use default props', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Layer/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with children', () => {

      it('should render children', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Layer>
            <span>Child</span>
          </Layer>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with className', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Layer className="Foo"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compClassName', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <Layer compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <Layer className="foo" compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <Layer className="foo" compClassName="layer"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

  })

})
