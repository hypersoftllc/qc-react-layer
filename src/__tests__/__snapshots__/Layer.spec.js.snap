// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layer rendered with children should render children 1`] = `
<div
  className="Block Layer"
>
  <span>
    Child
  </span>
</div>
`;

exports[`Layer rendered with className should render with expected class attribute 1`] = `
<div
  className="Block Layer Foo"
/>
`;

exports[`Layer rendered with compClassName should render with expected class attribute 1`] = `
<div
  className="Block"
/>
`;

exports[`Layer rendered with compClassName should render with expected class attribute 2`] = `
<div
  className="Block foo"
/>
`;

exports[`Layer rendered with compClassName should render with expected class attribute 3`] = `
<div
  className="Block layer foo"
/>
`;

exports[`Layer rendered with no props should use default props 1`] = `
<div
  className="Block Layer"
/>
`;
